pkg/processor/runtime/python/py/_nuclio_wrapper.py

```diff
@@ -212,47 +223,30 @@ async def _initialize_context(self):
                 raise
 
     def _register_to_signal(self):
-        signal.signal(signal.SIGUSR1, self._on_termination_signal)
-        signal.signal(signal.SIGUSR2, self._on_drain_signal)
-
-    def _on_drain_signal(self, signal_number, frame):
-        self._logger.debug_with('Received signal, calling draining callback',
-                                signal=signal.Signals(signal_number).name)
+        on_termination_signal = functools.partial(self._on_termination_signal, "SIGUSR1")
+        on_drain_signal = functools.partial(self._on_drain_signal, "SIGUSR2")
 
-        if self._is_waiting_for_event:
-            self._logger.debug('Wrapper is waiting for an event, calling drain handler')
+        asyncio.get_running_loop().add_signal_handler(signal.SIGUSR1, on_termination_signal)
+        asyncio.get_running_loop().add_signal_handler(signal.SIGUSR2, on_drain_signal)
 
-            # call the drain handler here as the event loop is stuck waiting for an event
-            self._call_drain_handler()
-        else:
-            self._logger.debug('Wrapper is handling an event, setting drain flag to true')
-
-            # set the flag to true so the event loop will call the drain handler
-            # after the current event is handled
-            self._is_drain_needed = True
-
-    def _on_termination_signal(self, signal_number, frame):
-        self._logger.debug_with('Received signal, calling termination callback',
-                                signal=signal.Signals(signal_number).name)
-
-        if self._is_waiting_for_event:
-            self._logger.debug('Wrapper is waiting for an event, calling termination handler')
-
-            # call the termination handler here as the event loop is stuck waiting for an event
-            self._call_termination_handler()
-        else:
-            self._logger.debug('Wrapper is handling an event, setting termination flag to true')
+    def _on_drain_signal(self, signal_name):
+        self._logger.debug_with('Received signal, calling draining callback', signal=signal_name)
+        self._is_drain_needed = True
+        if self._event_message_length_task:
+            self._event_message_length_task.cancel()
```

Comment:
Explain in a comment that we do this to unblock the wait for an event, so the drain callback would be invoked form the main loop.